types/wlr_buffer.c

```diff
@@ -254,7 +228,7 @@ struct wlr_client_buffer *wlr_client_buffer_import(
 	wlr_buffer_lock(&client_buffer->base);
 	wlr_buffer_drop(&client_buffer->base);
 
-	return client_buffer;
+	return &client_buffer->base;
```

Comment:
Returning the client buffer here is not okay. In the case of a `wl_shm` buffer and the GLES2 renderer, the buffer will be uploaded each time `wlr_buffer_import` is called. We should do this a single time per commit. but this patch does it more often since this function is used to get the buffer size.

Reply:
Note, buffer upload == buffer copy == slow :P

Reply:
I see what I got wrong, made a few changes and the wlr_buffer is imported only once now.